core/nuxeo-document.js

```diff
@@ -218,6 +225,39 @@ import './nuxeo-resource.js';
       }
       return path;
     }
+
+    appendClientReason(prop) {
+      prop.viewData = `${prop.data}&clientReason=view`;
+      prop.downloadData = `${prop.data}&clientReason=download`;
+    }
+
+    setDocumentViewDownloadProp() {
+      const documentProps = [];
+      if (this.documentData && this.documentData.properties) {
+        documentProps.push(
+          this.documentData.properties['file:content'],
+          this.documentData.properties['files:files'],
+          this.documentData.properties['picture:views'],
+          this.documentData.properties['vid:transcodedVideos'],
+          this.documentData.properties['vid:storyboard'],
+        );
+        documentProps.forEach((docProp) => {
+          if (docProp) {
+            if (Array.isArray(docProp)) {
+              docProp.forEach((prop) => {
+                if (prop.file) {
+                  this.appendClientReason(prop.file);
+                } else {
+                  this.appendClientReason(prop.content);
```

Comment:
Change this statements
if (prop.file) {
this.appendClientReason(prop.file);
} else {
this.appendClientReason(prop.content);
}
to below optimized code
this.appendClientReason(prop.file ? prop.file: prop.content);